code/game/objects/items/devices/scanners.dm

```diff
@@ -439,30 +444,29 @@ GAS ANALYZER
 
 		if(ongoing_weather)
 			if((ongoing_weather.stage == MAIN_STAGE) || (ongoing_weather.stage == WIND_DOWN_STAGE))
-				to_chat(user, "<span class='warning'>[src] can't trace anything while the storm is [ongoing_weather.stage == MAIN_STAGE ? "already here!" : "winding down."]</span>")
+				to_chat(user, "<span class='warning'>[src]'s barometer function can't trace anything while the storm is [ongoing_weather.stage == MAIN_STAGE ? "already here!" : "winding down."]</span>")
 				return
 
 			var/time = butchertime((ongoing_weather.next_hit_time - world.time)/10)
-			to_chat(user, "<span class='notice'>The next [ongoing_weather] will hit in [round(time)] seconds.</span>")
+			to_chat(user, "<span class='notice'>The next [ongoing_weather] will hit in [DisplayTimeText(round(time)*10)] seconds.</span>")
 			if(ongoing_weather.aesthetic)
-				to_chat(user, "<span class='warning'>[src] says that the next storm will breeze on by.</span>")
-		else if(user_area.outdoors)
+				to_chat(user, "<span class='warning'>[src]'s barometer function says that the next storm will breeze on by.</span>")
+		else
 			var/next_hit = SSweather.next_hit_by_zlevel["[T.z]"]
 			var/fixed = next_hit ? next_hit - world.time : -1
 			if(fixed < 0)
-				to_chat(user, "<span class='warning'>[src] was unable to trace any weather patterns.</span>")
+				to_chat(user, "<span class='warning'>[src]'s barometer function was unable to trace any weather patterns.</span>")
 			else
 				fixed = butchertime(round(fixed / 10))
-				to_chat(user, "<span class='warning'>A storm will land in approximately [fixed] seconds.</span>")
-		else
-			to_chat(user, "<span class='warning'>[src]'s barometer function won't work indoors!</span>")
+				to_chat(user, "<span class='warning'>[src]'s barometer function says a storm will land in approximately [fixed] seconds.</span>")
 		cooldown = TRUE
-		addtimer(src, /obj/item/device/analyzer/proc/ping, cooldown_time)
+		addtimer(CALLBACK(src,/obj/item/device/analyzer/proc/ping), cooldown_time)
+
 
 /obj/item/device/analyzer/proc/ping()
 	if(isliving(loc))
 		var/mob/living/L = loc
-		to_chat(L, "<span class='notice'>[src] is ready!</span>")
+		to_chat(L, "<span class='notice'>[src]'s barometer function is ready!</span>")
 	playsound(src, 'sound/machines/click.ogg', 100)
 	cooldown = FALSE
 

```

Comment:
Change butcher time's returns to use DisplayTimeText(blahblah)

Reply:
ok i guess